api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -285,8 +294,8 @@ func (s *SolutionManager) cleanupHeartbeat(ctx context.Context, id string, names
 }
 
 func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.DeploymentSpec, remove bool, namespace string, targetName string) (model.SummarySpec, error) {
-	lock.Lock()
-	defer lock.Unlock()
+	s.KeyLockProvider.Lock(namespace + "&&" + deployment.Instance.ObjectMeta.Name) // && used as split character
```

Comment:
Please write a util function to generate the lock key, in case we need to refer to same lock in other places.